Stratis.Bitcoin/Features/RPC/Controllers/FullNodeController.cs

```diff
@@ -110,6 +117,28 @@ public GetInfoModel GetInfo()
             return model;
         }
 
+        /// <summary>
+        /// Implements getblockheader RPC call
+        /// </summary>
+        /// <param name="hash">hash of block</param>
+        /// <param name="isJsonFormat">indicates whether to provide data in Json or binary format</param>
+        /// <returns>The block header rpc format</returns>
+        [ActionName("getblockheader")]
+        public BlockHeaderModel GetBlockHeader(string hash, bool isJsonFormat = true)
+        {
+            Guard.NotNull(this.Chain, nameof(this.Chain));
```

Comment:
The Guard method is only really used to check parameters.
So here you could have Guard.NotEmpty(hash, nameof(hash));
To check that the chain is not null, a simple if(this.chain == null) is sufficient (or != null depending on how you want to do this).